### PR TITLE
nbgrader 0.9.1 (manual, add maintainer)

### PR DESCRIPTION
**Pull request by bollwyvl**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

References:
- includes #63
- fixes #65

Notes:
- [x] update all deps
- [x] add `pip check`
- [x] use `jupyter` CLI to validate 
  - [ ] lab extensions (skipped due to https://github.com/jupyter/nbgrader/issues/1826) 
  - [x] server extensions
- [x] add bot inspection for future PRs
- [x] add self as maintainer

### CODE REVIEW

**Comment by bollwyvl** [comment]
@conda-forge/nbgrader ready for review

**Comment by brichet** [comment]
Thanks @bollwyvl.
I'm wondering how to manage versions with different dependencies in conda-forge.
The 0.8.5 has also been released (https://github.com/jupyter/nbgrader/releases/tag/v0.8.5), and it should be nice to provide it in conda-forge too (for people using lab 3 or notebook 6).

**Comment by bollwyvl** [comment]
Any branch on _this_ repo ships packages: so the recommendation is usually to have `main` be the latest release, with historical branches. Only `main` will get bot-created PRs, however. We've done this for e.g:
- [jupyterlab 3.x](https://github.com/conda-forge/jupyterlab-feedstock/tree/3.x)
- [ipywidgets 7.x](https://github.com/conda-forge/ipywidgets-feedstock/tree/7.x)

If this is merged, I can get a `0.8.x` set up pretty quickly (with the tests from this PR, etc. which become more relevant for historic branches vs latest-of-everything-not-pinned).

**Comment by brichet** [comment]
It should be OK now with version `0.9.1`: https://github.com/jupyter/nbgrader/releases/tag/v0.9.1

**Comment by bollwyvl** [comment]
Sure, I'll update this PR...

**Comment by bollwyvl** [comment]
@conda-forge-admin please rerender

**Comment by bollwyvl** [comment]
@conda-forge/nbgrader ready for review

**Comment by bollwyvl** [comment]
I've created https://github.com/conda-forge/nbgrader-feedstock/pull/66

It's still in draft, as:
- this PR should be merged
- a `0.8.x` branch needs to be created to target

**Comment by bollwyvl** [comment]
@conda-forge/nbgrader ready for review... I won't be able to do anything further this weekend, but could also handle #66 if this gets merged relatively soon today.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,59 +1,78 @@
...
+    - grep -iE "nbgrader.*OK.*nbgrader" labextensions
 
 about:
   home: https://github.com/jupyter/nbgrader
   license: BSD-3-Clause
   license_family: BSD
-  license_file: LICENSE
+  license_file:
+    - LICENSE
+    - nbgrader/labextension/static/third-party-licenses.json
```
Comment (brichet) [inline recipe/meta.yaml:73]: I assume the licence file now will be picked up from there ?
I don't know what *nbgrader/labextension/static/third-party-licenses.json* refer to.

**Review by brichet — APPROVED** [review]
I'm not maintainer of this repo but it looks good to me, at least for the dependencies and tests.

File: recipe/meta.yaml (hunk 1 above)
Comment (bollwyvl) [reply recipe/meta.yaml:73]: > I don't know what nbgrader/labextension/static/third-party-licenses.json refer to.

It's a collection of all declared  `LICENSE(.*)` files for all of the stuff that may end up being served to the client, as found by `webpack` during `jupyter labextension build`. 

A user can see thesse licenses in JupyterLab by going to _Help -> Licenses_ and looking at the `nbgrader` bundle.

Comment (brichet) [reply recipe/meta.yaml:73]: Thanks for clarification

**Comment by bollwyvl** [comment]
@conda-forge/nbgrader full week out now, folks, and the upstream has done a couple _releases_ to get this happy. please consider reviewing!

**Comment by bollwyvl** [comment]
@conda-forge/core a week has elapsed, with no feedback from maintainers (but lots of help from upstream developers :heart:).

This PR adds a new maintainer, and improves the overall robustness of the recipe. Please consider merging.

**Comment by bollwyvl** [comment]
Thanks! :heart_eyes:

**Pull request merged**